presto-hive/src/main/java/io/prestosql/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -321,4 +324,24 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
                     realToLongDecimal(fromType.getLong(block, position), toType.getPrecision(), toType.getScale()));
         }
     }
+
+    public static Function<Block, Block> createDecimalToVarcharCoercer(DecimalType fromType, VarcharType toType)
+    {
+        return new DecimalToVarcharCoercer(fromType, toType);
+    }
+
+    private static class DecimalToVarcharCoercer
+            extends TypeCoercer<DecimalType, VarcharType>
+    {
+        public DecimalToVarcharCoercer(DecimalType fromType, VarcharType toType)
+        {
+            super(fromType, toType);
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            toType.writeSlice(blockBuilder, utf8Slice(Decimals.toString(fromType.getSlice(block, position), fromType.getScale())));
```

Comment:
What if the slice is longer than toType.getLength()?

Reply:
From digging deeper into the `writeSlice` method eventually all calls end up at `VariableWidthBlockBuilder#writeBytes()` which ends up calling `SliceOutput.writeBytes`. Each of the implementors of that method look like they are resizing the underlying buffer until the entire slice has been written.

So I don't think this should have any impact. Correct me if I'm wrong or am misreading/misunderstanding something.

Reply:
@findepi Is my reading/understanding of the correct?